### DISCUSSION TITLE
Why are some controls like a button and progressBar not displayed when they are added to the VisualChildren of a Control?

### DISCUSSION THREAD

**Discussion by LZD121** [AvaloniaUI/Avalonia#15434] [Q&A]
This is my test code:
```
public class MyControl: Control
 {
    private readonly Button _button;

    public MyControl()
    {
        VisualChildren.Add(new Grid { Background = Brushes.Green });
        VisualChildren.Add(new Ellipse
        {
            Width = 100,
            Height = 100,
            Fill = Brushes.Red
        });

        _button = new Button
        {
            Content = "Button Test",
            Foreground = Brushes.White,
            Background = Brushes.Yellow
        };
        VisualChildren.Add(_button);
    }

    protected override Size ArrangeOverride(Size finalSize)
    {
        _button.Arrange(new Rect(100, 100, 50, 60));

        return base.ArrangeOverride(finalSize);
    }
}
```

The result is that Grid and Ellipse are displayed, but the button is not.
![image](https://github.com/AvaloniaUI/Avalonia/assets/59716053/4059eaf8-be4f-4e02-9cfd-61886acae52f)

But on Visual Tree I can find it:
![image](https://github.com/AvaloniaUI/Avalonia/assets/59716053/4b42c50b-6cce-47e9-a074-bcae946a01d8)

Why is this happening? How can I add a button and make it visible?

**Comment by jp2masa** [comment]
I think you also need to implement `MeasureOverride`. `Button` is a `TemplatedControl`, which applies the template when measuring.

**Comment by timunie** [comment]
I'd not suggest to manipulate the Visual or Logical tree on your own. Better to use a Panel to place Children.
